src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -1764,6 +1764,110 @@ JSC_DEFINE_HOST_FUNCTION(jsSQLStatementFcntlFunction, (JSC::JSGlobalObject * lex
 
     return JSValue::encode(jsNumber(statusCode));
 }
+typedef struct {
+    Strong<JSC::JSGlobalObject> globalObject;
+    Strong<JSC::JSFunction> callback;
+} UpdateHookData;
+
+UpdateHookData * data = nullptr;
+
+void on_update_hook(void * dataPtr, int type, const char * dbName, const char * tableName, sqlite_int64 rowId) {
+    UpdateHookData* data = (UpdateHookData *)dataPtr;
+
+    auto globalObject = data->globalObject.get();
+    auto callback = data->callback.get();
+
+    JSValue typeValue = jsNumber(type);
+    JSValue dbNameValue = jsString(globalObject->vm(), makeString(ASCIILiteral::fromLiteralUnsafe(dbName)));
+    JSValue tableNameValue = jsString(globalObject->vm(), makeString(ASCIILiteral::fromLiteralUnsafe(tableName)));
```

Comment:
When creating a string from a UTF-8 data source, we need to use `String::fromUTF8` instead. `ASCIILiteral` can only be used for constant strings like when you do `"foo"_s` .